baselines/feddebug/README.md

```diff
@@ -0,0 +1,221 @@
+
+```bash
+python -m feddebug.main device=cpu
+```  
+
+Output of the last round will show the FedDebug output as follows:
+
+```log
+... 
+[2024-10-24 12:25:48,758][flwr][INFO] - ***FedDebug Output Round 5 ***
+[2024-10-24 12:25:48,758][flwr][INFO] - True Malicious Clients (Ground Truth) = ['0']
+[2024-10-24 12:25:48,758][flwr][INFO] - Total Random Inputs = 10
+[2024-10-24 12:25:48,758][flwr][INFO] - Predicted Malicious Clients = {'0': 1.0}
+[2024-10-24 12:25:48,758][flwr][INFO] - FedDebug Localization Accuracy = 100.0
+[2024-10-24 12:25:49,577][flwr][INFO] - fit progress: (5, 0.00015518503449857236, {'accuracy': 0.978, 'loss': 0.00015518503449857236, 'round': 5}, 39.02993568999227)
+[2024-10-24 12:25:49,577][flwr][INFO] - configure_evaluate: no clients selected, skipping evaluation
+[2024-10-24 12:25:49,577][flwr][INFO] - 
+[2024-10-24 12:25:49,577][flwr][INFO] - [SUMMARY]
+...
+
+```
+It predicts the malicious client(s) with 100% accuracy. `Predicted Malicious Clients = {'0': 1.0}`  means that client-0 is predicted as the malicious client with 1.0 probability. It will also generate a graph `iid-lenet-mnist.png` as shown below: 
+![FedDebug Malicious Client Localization IID-LeNet-MNIST](figures/iid-lenet-mnist.png)
+
+### a. Multiple Malicious Clients
+To test the localization of multiple malicious clients, you can change the `total_malicious_clients`. Total Time Taken: 46.7 seconds.
+
+```bash
+python -m feddebug.main device=cpu total_malicious_clients=2 dataset.name=cifar10
+```
+In this scenario, clients 0 and 1 are now malicious. The output will show the FedDebug output as follows:
+
+```log
+...
+[2024-10-24 12:28:14,125][flwr][INFO] - ***FedDebug Output Round 5 ***
+[2024-10-24 12:28:14,125][flwr][INFO] - True Malicious Clients (Ground Truth) = ['0', '1']
+[2024-10-24 12:28:14,125][flwr][INFO] - Total Random Inputs = 10
+[2024-10-24 12:28:14,125][flwr][INFO] - Predicted Malicious Clients = {'0': 1.0, '1': 1.0}
+[2024-10-24 12:28:14,125][flwr][INFO] - FedDebug Localization Accuracy = 100.0
+[2024-10-24 12:28:15,148][flwr][INFO] - fit progress: (5, 0.003398957598209381, {'accuracy': 0.4151, 'loss': 0.003398957598209381, 'round': 5}, 35.81892481799878)
+[2024-10-24 12:28:15,148][flwr][INFO] - configure_evaluate: no clients selected, skipping evaluation
+[2024-10-24 12:28:15,148][flwr][INFO] - 
+[2024-10-24 12:28:15,148][flwr][INFO] - [SUMMARY]
+...
+```
+FedDebug predicts the malicious clients with 100% accuracy. `Predicted Malicious Clients = {'0': 1.0, '1': 1.0}` means that clients 0 and 1 are predicted as the malicious clients with 1.0 probability. It will also generate a graph `iid-lenet-cifar10.png` as shown below:
+
+![FedDebug Malicious Client Localization IID-LeNet-CIFAR10](figures/iid-lenet-cifar10.png)
+
+
+### b. Changing the Model and Device
+To run the experiments with ResNet and Cuda with Non-IID distribution you can run the following command. Total Time Taken: 84 seconds.
```

Comment:
In your `initialize_model` you have support to a few models. Would it be worth adding here a note to emphasize it ? It's great that's so easy to change the model.

> [!NOTE]
> You can run feddebug with anymodel in `<path-to-file/code-line>`.... .

Reply:
Thank you, I have added the note.